nixos/modules/security/auditd.nix

```diff
@@ -7,6 +7,77 @@
 let
   cfg = config.security.auditd;
 
+  plugin_conf = lib.types.submodule {
+    options = {
+      active = lib.mkEnableOption "whether to enable this plugin";
+      direction = lib.mkOption {
+        type = lib.types.enum [
+          "in"
+          "out"
+        ];
+        default = "out";
+        description = ''
+          The option is dictated by the plugin.  In or out are the only choices.
+          You cannot make a plugin operate in a way it wasn't  designed just by
+          changing this option. This option is to give a clue to the event dispatcher
+          about which direction events flow. NOTE: inbound events are not supported yet.
+        '';
+      };
+      path = lib.mkOption {
+        type = lib.types.path;
+        description = "This is the absolute path to the plugin executable.";
+      };
+      type = lib.mkOption {
+        type = lib.types.enum [ "always" ];
+        default = "always";
+        description = ''
+          This  tells the dispatcher how the plugin wants to be run. There is only
+          one valid option, `always`, which means the plugin is external and should
+          always be run. The default is always since there are no more builtin plugins.
+        '';
+      };
+      args = lib.mkOption {
+        type = lib.types.nullOr (lib.types.listOf lib.types.nonEmptyStr);
+        default = null;
+        description = ''
+          This allows you to pass arguments to the child program.
+          Generally plugins do not take arguments and have their own
+          config file that instructs them how they should be configured.
+        '';
+      };
+      format = lib.mkOption {
+        type = lib.types.enum [
+          "binary"
+          "string"
+        ];
+        default = "string";
+        description = ''
+          Binary passes the data exactly as the audit event dispatcher gets it from
+          the audit  daemon. The string option tells the dispatcher to completely change
+          the event into a string suitable for parsing with the audit parsing library.
+        '';
+      };
+      config = lib.mkOption {
+        type = lib.types.nullOr (
+          lib.types.submodule {
+            freeformType =
+              with lib.types;
+              attrsOf (
+                nullOr (oneOf [
+                  bool
+                  nonEmptyStr
+                  path
+                  int
+                ])
+              );
```

Comment:
Does this type need to be kept in sync with the primary configuration? If so, can I suggest referencing it from the let block?